### DISCUSSION TITLE
Dynamic width for Modal

### DISCUSSION THREAD

**Discussion by edoardo** [dhis2/notes#305] [Q&A] [answered]
`Modal` currently has `small` and `large` boolean props to control the width, which I understand they simply set the width to a fixed size in CSS (800px for large).

We have requests for wider Modals that can use more horizontal space when using wide screens.

Can there be another option for the width? A dynamic width options would be nice, where the `maxWidth` (which is already set?) is used but without specify a fixed `width`.

**Comment by varl** [comment, marked as answer]
Yeah I think this can be discussed. I've added it to Jira here: https://jira.dhis2.org/browse/LIBS-221

**Reply by varl** [reply]
In the related LIBS-2 issue (custom size) we have solved it with a `fluid` prop.

It allows for both use cases:

1) The Modal is dynamic in size if the `fluid` prop is passed, based on the content.

2) By using the pattern `ModalContent > div` or `ModalContent > Box` and setting a size to the `div`/`Box`, a custom size can be accomplished.

**Reply by varl** [reply]
https://jira.dhis2.org/browse/LIBS-2?focusedCommentId=45849&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-45849

**Reply by varl** [reply]
PR: https://github.com/dhis2/ui/pull/838
